CMakeLists.txt

```diff
@@ -55,45 +59,49 @@ endif()
 ####################################################
 # Build Options
 ####################################################
-SET(BUILD_TESTS YES CACHE BOOL "Build test applications")
+
+option(BUILD_TESTS "Build test applications" YES)
 
 if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/wolfssl/options.h")
     # Copy generated ./options.h
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/wolfssl/options.h 
-                   ${CMAKE_CURRENT_BINARY_DIR}/user_settings.h)
+                   ${CMAKE_CURRENT_SOURCE_DIR}/user_settings.h)
 else()
    # Use template
    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/wolfssl/options.h.in 
-                  ${CMAKE_CURRENT_BINARY_DIR}/user_settings.h)
+                  ${CMAKE_CURRENT_SOURCE_DIR}/user_settings.h)
 endif()
 
 add_definitions(-DWOLFSSL_USER_SETTINGS)
 add_definitions(-DWOLFSSL_IGNORE_FILE_WARN)
-if(CMAKE_HAVE_PTHREAD_H)
+if(CMAKE_USE_PTHREADS_INIT)
   add_definitions(-DHAVE_PTHREAD)
 endif()
 
 ####################################################
-# Source Files
+# Library Target and Source Files
 ####################################################
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/.)
-include_directories(${CMAKE_CURRENT_BINARY_DIR}/.)
 
 file(GLOB LIB_SOURCE_FILES
```

Comment:
In a future commit, I would propose we stop globbing source files for the reasons outlined here: https://stackoverflow.com/questions/1027247

Reply:
That's fine with me. It's a bit of additional work when new files are added, but that is okay.